src/sc/sidechaintypes.h

```diff
@@ -4,19 +4,278 @@
+};
+
+typedef CFieldElement ScConstant;
+////////////////////////////// End of Field types //////////////////////////////
+
+/////////////////////// libzendoomc namespace definitions //////////////////////
+namespace libzendoomc {
+    typedef base_blob<SC_PROOF_SIZE * 8> ScProof;
+
+    /* Check if scProof is a valid zendoo-mc-cryptolib's sc_proof */
+    bool IsValidScProof(const ScProof& scProof);
+
+    typedef base_blob<SC_VK_SIZE * 8> ScVk;
+
+    /* Check if scVk is a valid zendoo-mc-cryptolib's sc_vk */
+    bool IsValidScVk(const ScVk& scVk);
+
+    /* Convert to std::string a zendoo-mc-cryptolib Error. Useful for logging */
+    std::string ToString(Error err);
+}
+/////////////////// end of libzendoomc namespace definitions ///////////////////
+
+////////////////////////////// Custom Config types //////////////////////////////
+class CustomCertificateFieldConfig
+{
+public:
+    CustomCertificateFieldConfig() = default;
+    virtual ~CustomCertificateFieldConfig() = default;
+    virtual bool IsValid() const = 0;
+};
+
+class FieldElementCertificateFieldConfig : public CustomCertificateFieldConfig
+{
+private:
+    int32_t nBits;
+
+public:
+    FieldElementCertificateFieldConfig(int32_t nBitsIn);
+    FieldElementCertificateFieldConfig(const FieldElementCertificateFieldConfig& rhs) = default;
+    ~FieldElementCertificateFieldConfig() = default;
+
+    //For serialization only, which requires the default ctor. No checkValid call here
+    FieldElementCertificateFieldConfig(): CustomCertificateFieldConfig(), nBits(0) {}
+
+    bool IsValid() const override final;
+
+    ADD_SERIALIZE_METHODS;
+    template <typename Stream, typename Operation>
+    inline void SerializationOp(Stream& s, Operation ser_action, int nType, int nVersion) {
+        READWRITE(nBits);
+    }
 
-#include<sc/proofverifier.h>
+    int32_t getBitSize() const;
 
-//------------------------------------------------------------------------------------
-class CTxForwardTransferOut;
+    bool operator==(const FieldElementCertificateFieldConfig& rhs) const {
+        return (this->nBits == rhs.nBits);
+    }
+
+    bool operator!=(const FieldElementCertificateFieldConfig& rhs) const {
+        return !(*this == rhs);
+    }
+
+    friend std::ostream& operator<<(std::ostream& os, const FieldElementCertificateFieldConfig& r) {
+        os << r.nBits;
+        return os;
+    }
+};
+
+class BitVectorCertificateFieldConfig : public CustomCertificateFieldConfig
+{
+private:
+    int32_t bitVectorSizeBits;
+    int32_t maxCompressedSizeBytes;
+
+public:
+    BitVectorCertificateFieldConfig(int32_t bitVectorSizeBits, int32_t maxCompressedSizeBytes);
+    ~BitVectorCertificateFieldConfig() = default;
+
+    //for serialization only, which requires the default ctor. No checkValid call here
+    BitVectorCertificateFieldConfig(): CustomCertificateFieldConfig(), bitVectorSizeBits(-1), maxCompressedSizeBytes(-1) {}
+
+    static const int32_t MAX_BIT_VECTOR_SIZE_BITS = 1000192;
```

Comment:
change to 1000760 that is both modulus 254 ("fully available" bits in a field element whose modulus is 255) and modulus 8 (to be easily managed in serialization/deserialization)

Reply:
Done